src/controllers/__tests__/user.test.js

```diff
@@ -0,0 +1,194 @@
+    test('Should send the correct user in the response', async () => {
+      const response = await request(server).get(`/api/users/${userId}`);
+
+      const responseBody = response.body;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(200);
+      expect(responseBody).toMatchObject(expectedValidUser);
+    });
+
+    test('Should response with an error message when requested user ID does not exist', async () => {
+      const response = await request(server).get(
+        `/api/users/${notExistingUserId}`
+      );
+
+      const responseBody = response.body;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBe(
+        "The user with the specified ID wasn't found"
+      );
+    });
+
+    test('Should response with an error message when requested user ID is not valid', async () => {
+      const response = await request(server).get(`/api/users/${invalidId}`);
+      const responseBody = response.body;
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBe('Requested user ID is not valid!');
+    });
+  });
+
+  describe('PUT /api/users/:id', () => {
+    test('Should update a user and save it to database', async () => {
+      const response = await request(server)
+        .put(`/api/users/${userId}`)
+        .send({ username: 'updated username' });
+
+      const responseBody = response.body;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(200);
+      expect(responseBody).toMatchObject(updatedUser);
+    });
+
+    test('Should response with an error message when requested user ID does not exist', async () => {
+      const response = await request(server).get(
+        `/api/users/${notExistingUserId}`
+      );
+
+      const responseBody = response.body;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBe(
+        "The user with the specified ID wasn't found"
+      );
+    });
+    test('Should response with an error message when requested user ID is not valid', async () => {
+      const response = await request(server).get(`/api/users/${invalidId}`);
+      const responseBody = response.body;
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBe('Requested user ID is not valid!');
+    });
+  });
+
+  describe('DELETE /api/users/:id', () => {
+    test('Should delete the correct user from the database', async () => {
+      const response = await request(server).delete(`/api/users/${userId}`);
+      expect(response.statusCode).toBe(204);
```

Comment:
Here you're only checking for the response status code.
You can also run the GET request again and confirm that the user is deleted from the DB.